src/esssans/common.py

```diff
@@ -64,12 +64,19 @@ def mask_range(
     lu = sc.DataArray(data=mask.data, coords={dim: edges})
     if da.bins is not None:
         if dim not in da.coords:
-            da = da.bin({dim: 1})
-        new_bins = sc.array(
-            dims=[dim],
-            values=np.union1d(edges.values, da.coords[dim].values),
-            unit=edges.unit,
-        )
+            underlying = da.bins.coords[dim]
+            new_bins = np.union1d(
+                edges.values,
+                np.array(
+                    [
+                        underlying.min().value,
+                        np.nextafter(underlying.max().value, np.inf),
+                    ]
+                ),
+            )
+        else:
+            new_bins = np.union1d(edges.values, da.coords[dim].values)
```

Comment:
What if the units do not match?

Reply:
I think it's actually ok in this particular case, because below, we do
```Py
new_bins = sc.array(dims=[dim], values=new_bins, unit=edges.unit)
out = da.bin({dim: new_bins})
```
So the bad unit would be caught by the `da.bin` (I think?).

However, it's more a matter of luck than anything else. I will add an explicit check.

Reply:
I think we should add a conversion (or check) of the mask edges to the correct unit close to the beginning of the function, where also other checks are performed. Otherwise we get a cryptic error from `sc.bin` here, or worse, someone will modify the code and the coincidental unit check will suddenly disappear.